proto/p4/config/p4info.proto

```diff
@@ -21,13 +21,11 @@ import "google/protobuf/any.proto";
 package p4.config;
 
 message P4Info {
-  repeated HeaderField header_fields = 1;
```

Comment:
In both cases you ultimately just have a string, right? You'll always need to have semantic knowledge of the P4 program (or at least, knowledge of some convention that the P4 program adheres to) to interpret them. Seems like whether we regard those strings as "header fields" or "match keys" doesn't change much in the end. (If I'm missing something, though, I'd love to know more!)